Add docs about crate level documentation support (#76289)
rust-lang/rust: src/doc/rustdoc/src/what-is-rustdoc.md

your code relies on. If our project used dependencies, we'd get
documentation for them as well!

## Crate level documentation
From the above examples, you can see that using the `///` syntax
documents the next item. But that can't be used when you are
documenting the entire crate, because there is nothing before the crate.
It is where everything is placed under. So, in order to document an entire
crate, you use this `//!` syntax. For example

``` rust
//! This is my first rust crate
```

> For more information about the `//!` syntax, see [the Book](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commenting-contained-items).

Reviewer: nit: I think this would be better on two lines so it's not as long.
```suggestion
For more information about the `//!` syntax, see [the Book].

[the Book]: https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commenting-contained-items
```

Author: So what should I do,? Which one should I resolve

Author: So what should I do,? Which one should I resolve

Reviewer: You can make multiple changes at once, you don't have to make them one at a time and ask between each :)

your code relies on. If our project used dependencies, we'd get
documentation for them as well!

## Crate level documentation
From the above examples, you can see that using the `///` syntax
documents the next item. But that can't be used when you are

> documenting the entire crate, because there is nothing before the crate.
> It is where everything is placed under. So, in order to document an entire
> crate, you use this `//!` syntax. For example

Reviewer: ```suggestion
documenting the entire crate, because nothing comes before it: it is the
root of the crate. So in order to document an entire crate, you need to use `//!` syntax. For example:
```

Reviewer: You marked this as resolved but didn't apply the suggestion.

Reviewer: You did it again 😆

Reviewer: You still have not resolved this comment. If you don't think this is a good change to make, please explain why instead of continuing to ignore it.

Reviewer: You still have not addressed this comment. If you don't think this is a good change to make, please explain why instead of ignoring it.

``` rust
//! This is my first rust crate
```

> Actually the `//!` syntax is used to document the item it is present
> in. So by using this at the crate root, we want to document the item it
> is present in, which is the crate itself. You can use it inside functions,
> structs as well.

Reviewer: ```suggestion
The `//!` syntax is used to document the item it is present in,
so when used at the crate root, it documents the crate itself.
You can use it anywhere that is inside a larger item, such as
`struct` implementations, function bodies, and modules.
```

Reviewer: Oh actually I missed that https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html?highlight=Document#commenting-contained-items already exists - we should link there rather than repeating the info.

Reviewer: You marked this as resolved but didn't apply the second suggestion.

Author: I am just doing it

Reviewer: Gotcha. Please wait to mark this as resolved until you've made the change in the future.

your code relies on. If our project used dependencies, we'd get
documentation for them as well!

> ## Outer and inner documentation

Reviewer 2: adding a space between this heading and the next paragraph would be good, to match the rest of the documents.

The `///` syntax is used to document the item present after it. That's why it is called
an outer documentation.
There is another syntax: `//!`, which is used to document the item it is present inside.

> It is called an inner documentation.

Reviewer 2: there are newlines after this sentence and the previous that should not be here.

It is often used when documenting the entire crate, because nothing comes before it: it is the
root of the crate. So in order to document an entire crate, you need to use `//!` syntax. For example:

``` rust
//! This is my first rust crate
```

When we use it in the crate root,

> we meant to say that we want to document the item it is present inside, which is the crate itself

Reviewer 2: also here

documentation for them as well!

## Outer and inner documentation

The `///` syntax is used to document the item present after it.
That's why it is called an outer documentation.
There is another syntax: `//!`, which is used to document the 
item it is present inside. It is called an inner documentation.
It is often used when documenting the entire crate,
because nothing comes before it: it is the root of the crate.
So in order to document an entire crate, you need to use `//!` syntax.
For example:

``` rust
//! This is my first rust crate
```

> When we use it in the crate root, we meant to say that we want to
> document the item it is present inside, which is the crate itself

Reviewer: One last little nit:
```suggestion
When used in the crate root, it documents the item it is inside,
which is the crate itself.
```
but other than that this looks good to me :)
